libbeat/dashboards/modify_json.go

```diff
@@ -46,25 +46,33 @@ func ReplaceIndexInIndexPattern(index string, content common.MapStr) common.MapS
 		return content
 	}
 
-	objects, ok := content["objects"].([]interface{})
-	if !ok {
-		return content
-	}
-
-	// change index pattern name
-	for i, object := range objects {
-		objectMap, ok := object.(map[string]interface{})
-		if !ok {
-			continue
+	if objectMaps, ok := content["objects"].([]common.MapStr); ok {
+		// change index pattern name
+		for i, objectMap := range objectMaps {
+
+			objectMap["id"] = index
+			if attributes, ok := objectMap["attributes"].(common.MapStr); ok {
+				attributes["title"] = index
+			}
+			objectMaps[i] = objectMap
 		}
-
-		objectMap["id"] = index
-		if attributes, ok := objectMap["attributes"].(map[string]interface{}); ok {
-			attributes["title"] = index
+		content["objects"] = objectMaps
+	} else if objects, ok := content["objects"].([]interface{}); ok {
+		// change index pattern name
+		for i, object := range objects {
+			objectMap, ok := object.(map[string]interface{})
+			if !ok {
+				continue
+			}
+
+			objectMap["id"] = index
+			if attributes, ok := objectMap["attributes"].(map[string]interface{}); ok {
+				attributes["title"] = index
+			}
+			objects[i] = objectMap
 		}
-		objects[i] = objectMap
+		content["objects"] = objects
 	}
-	content["objects"] = objects
```

Comment:
When dealing with external input and some implicitely assumed types, then we should validate and log if something doesn't match our expectations.

Maybe use this pattern:

```
  switch objects := content["objects"].(type) {
    case []interface{}:
        ...
    case []common.MapStr;
        ...
    default:
        // log error with type + contents e.g.
        logp.Errf("Unexpected object type %T, expected list of objects. Got: %#v", content["objects"], content["objects"])
  }
```


This if-else construct seems to normalize the object contents for further processing. Maybe we should error if we can't do so, but the field is present? How about moving the normalization into a separate function?

Reply:
Thanks, I've changed to a `switch`.

I initially tried separating the normalization from the rest of the logic. Unfortunately, it's not just a single type cast. The whole structure is nested, with each nested object being either `[]interface{}` or `[]common.MapStr`. So we'd need to recursively cast everything from one to the other, and maybe that's out of scope for this PR (which is essentially just fixing a regression bug)?